### ISSUE TITLE
Razor should generate fields for `@ref`

### ISSUE THREAD

**Issue by meziantou** [dotnet/aspnetcore#34826]
### Is your feature request related to a problem? Please describe.

When using `@ref` in a component, you need to manually create the corresponding fields:

```razor
<div @ref="element"></div>
<MyComponent @ref="component" />

@code {
    ElementReference element;
    MyComponent component;
}
```

### Describe the solution you'd like

The razor compiler should generate the fields automatically if it is not already declared

```razor
<div @ref="element"></div>
<MyComponent @ref="component" />

@code {
    // "element" is generated by the compiler
    async Task Sample() { await element.FocusAsync(); }
}
```

In the end, this would provide a similar experience to WPF with `x:name` or WinForms.

**Comment by campersau** [comment]
This was actually implemented in a 3.0.0-preview8 but got reverted in 3.0.0-preview9 because it didn't work. See https://github.com/aspnet/Announcements/issues/381

**Comment by javiercn** [comment]
@meziantou thanks for contacting us.

@campersau answer is correct here.
